lib/include/jxl/decode.h

```diff
@@ -913,6 +950,142 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderSetJPEGBuffer(JxlDecoder* dec,
+ * of a "brob" box. If false, automatically decompresses the box with brotli if
+ * it was compressed. This requires the decoder has been compiled with brotli
+ * support.
+ * @return JXL_DEC_ERROR if input was already set without releasing,
+ * JXL_DEC_SUCCESS otherwise
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderSetBoxBuffer(JxlDecoder* dec,
+                                                   uint8_t* data, size_t size,
+                                                   JXL_BOOL get_raw_box_data);
+
+/**
+ * Configures whether to get boxes in raw mode or in uncompressed
+ * mode.
+ *
+ * The default mode is non-raw (JXL_FALSE), where "brob" boxes are transparently
+ * decompressed. This setting can only be changed
+ * before decoding, or directly after a JXL_DEC_BOX event, and is preserved
+ * otherwise.
+ *
+ * @param dec decoder object
+ * @param enable_raw whether to enable or disable raw mode
+ * @return JXL_DEC_ERROR if input was already set without releasing,
+ * JXL_DEC_SUCCESS otherwise
+ */
+JXL_EXPORT void JxlDecoderSetBoxModeRaw(JxlDecoder* dec, JXL_BOOL enable_raw);
+
+/**
+ * Releases buffer which was provided with JxlDecoderSetBoxBuffer.
+ *
+ * Calling JxlDecoderReleaseBoxBuffer is required whenever
+ * a buffer is already set and a new buffer needs to be added with
+ * JxlDecoderSetBoxBuffer, but is not required before JxlDecoderDestroy or
+ * JxlDecoderReset.
+ *
+ * Calling JxlDecoderReleaseBoxBuffer when no buffer is set is
+ * not an error and returns 0.
+ *
+ * @param dec decoder object
+ * @return the amount of bytes the decoder has not yet written to of the data
+ * set by JxlDecoderSetBoxBuffer, or 0 if no buffer is set or
+ * JxlDecoderReleaseBoxBuffer was already called.
+ */
+JXL_EXPORT size_t JxlDecoderReleaseBoxBuffer(JxlDecoder* dec);
+
+/**
+ * Outputs the type of the current box, after a JXL_DEC_BOX event occured, as 4
+ * characters plus a null termination character. If the box is of type "brob",
+ * the underlying uncompressed box type is returned instead, not "brob".
+ *
+ * @param dec decoder object
+ * @param type buffer to copy the type into
+ * @return JXL_DEC_SUCCESS if the value is available, JXL_DEC_ERROR if not, for
+ *    example the JXL file does not use the container format.
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderGetBoxType(JxlDecoder* dec, char type[5]);
+
+/**
+ * Outputs whether the given type equals the type returned by
+ * JxlDecoderGetBoxType.
+ *
+ * @param dec decoder object
+ * @param type box type to compare with
+ * @return JXL_TRUE if the box type is available and equal to the given type,
+ * JXL_FALSE otherwise.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderBoxTypeEquals(JxlDecoder* dec, const char* type);
+
+/**
+ * Outputs the type of the current box, after a JXL_DEC_BOX event occured, as 4
+ * characters plus a null termination character. If the box is of type "brob",
+ * the underlying uncompressed box type is returned instead, not "brob".
```

Comment:
I'm pretty sure this is not what is intended.

Reply:
Not sure what was the non intended part, but reworded the sentence

Reply:
This is the documentation on the "BoxTypeRaw()" function, which was identical to the non-raw version - now the non-raw version was modified, but as far as I can see the documentation of the raw one is still incorrect.

Reply:
Ah, fixed